x-pack/platform/plugins/shared/alerting/server/rules_client/lib/extract_references.ts

```diff
@@ -20,9 +21,11 @@ export async function extractReferences<
   context: RulesClientContext,
   ruleType: UntypedNormalizedRuleType,
   ruleActions: NormalizedAlertActionWithGeneratedValues[],
-  ruleParams: Params
+  ruleParams: Params,
+  ruleArtifacts?: Artifact
```

Comment:
```suggestion
  ruleArtifacts: Artifact
```

As I mentioned in my above comment, I think we should make the `artifacts` required in the `extractReferences` to be sure we do not make the references empty when the rule clients use the function.

Reply:
See e625e734d3ee1a8975ca4d5a3af6fd4e8a8e6216.

Reply:
The initial change to make the artifacts required was done as part of this [commit](https://github.com/elastic/kibana/pull/216292/commits/6254e0f7550169eb4cdc2d1692ea21d8f65a3455). @cnasikas If you check the commit, you'll see a TODO I added for you to check.